MergerService/Runners/TaskRunner.cs

```diff
@@ -0,0 +1,129 @@
+using MergerLogic.Clients;
+using MergerLogic.Monitoring.Metrics;
+using MergerService.Models.Tasks;
+using MergerService.Utils;
+using System.Diagnostics;
+using System.Net;
+using System.Reflection;
+
+namespace MergerService.Runners
+{
+    public class TaskRunner : ITaskRunner
+    {
+        private readonly IHeartbeatClient _heartbeatClient;
+        private readonly IMetricsProvider _metricsProvider;
+        private readonly ITaskExecutor _taskExecutor;
+        private readonly ITaskUtils _taskUtils;
+        private readonly IJobUtils _jobUtils;
+        private readonly ILogger _logger;
+        private readonly MergerLogic.Utils.IConfigurationManager _configurationManager;
+
+        public TaskRunner(ITaskExecutor taskExecutor, IJobUtils jobUtils, ILogger<TaskRunner> logger,
+            ITaskUtils taskUtils, IHeartbeatClient heartbeatClient, IMetricsProvider metricsProvider,
+            MergerLogic.Utils.IConfigurationManager configurationManager)
+        {
+            this._taskUtils = taskUtils;
+            this._heartbeatClient = heartbeatClient;
+            this._metricsProvider = metricsProvider;
+            this._taskExecutor = taskExecutor;
+            this._jobUtils = jobUtils;
+            this._logger = logger;
+            this._configurationManager = configurationManager;
+        }
+
+        public List<KeyValuePair<string, string>> BuildTypeList()
+        {
+            var taskTypes = this._configurationManager.GetChildren("TASK", "types");
+
+            List<KeyValuePair<string, string>> values = new List<KeyValuePair<string, string>>();
+
+            foreach (var pair in taskTypes)
+            {
+                var jobType = pair.GetValue<string>("JobType");
+                var taskType = pair.GetValue<string>("taskType");
+                values.Add(new KeyValuePair<string, string>(jobType, taskType));
+            }
+
+            return values;
+        }
+
+        public bool FetchAndRunTasks(KeyValuePair<string, string> jobTaskTypesPair)
```

Comment:
Please split to two, one for fetching a task and another for running a task.

Reply:
done